### PR TITLE
Arch setup

### PR DESCRIPTION
**Pull request by ShobDeloitte**

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
```
Comment (asdheeraj) [inline .idea/vcs.xml]: you have to delete the .idea folder in your project before pushing the code to the repo. Ensure the files that are reflected in the git repo does not reflect any idea file

File: .idea/gradle.xml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="GradleMigrationSettings" migrationVersion="1" />
   <component name="GradleSettings">
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
```
Comment (asdheeraj) [inline .idea/gradle.xml]: this file should not be pushed to the repo

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/activity/LoginActivity.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.dsu.pocapplication.news.ui.login.activity
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/pocapplication/news/ui/login/activity/LoginActivity.kt]: there is no requirement of package "activity" inside login

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/fragments/LoginFragment.kt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.pocapplication.news.ui.login.fragments
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/pocapplication/news/ui/login/fragments/LoginFragment.kt]: there is no requirement of package "fragments" inside login

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/repository/LoginRepository.kt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.pocapplication.news.ui.login.repository
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/pocapplication/news/ui/login/repository/LoginRepository.kt]: repository package should not be a part of ui

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/viewmodel/LoginViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.pocapplication.news.ui.login.viewmodel
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/pocapplication/news/ui/login/viewmodel/LoginViewModel.kt]: there is no requirement of package "viewmodel" inside login

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/repository/LoginRepository.kt (hunk 5 above)
Comment (asdheeraj) [reply app/src/main/java/com/dsu/pocapplication/news/ui/login/repository/LoginRepository.kt]: Please read and understand about repository pattern initially: https://developer.android.com/jetpack/docs/guide

**Review by asdheeraj — COMMENTED** [review]
Please address the comments and kindly read about the repository pattern first: https://developer.android.com/jetpack/docs/guide

**Comment by asdheeraj** [comment]
Please keep "squashing" ur commits and keep resolving conflicts. If you are unaware of this, please understand how "git" works. Here is a start: https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing

**Comment by asdheeraj** [comment]
Please have a look into this regarding gitignore: https://github.com/intellij-rust/intellij-rust/issues/1570

File: .idea/vcs.xml (hunk 1 above)
Comment (ShobDeloitte) [reply .idea/vcs.xml]: deleted

File: .idea/gradle.xml (hunk 2 above)
Comment (ShobDeloitte) [reply .idea/gradle.xml]: done

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/activity/LoginActivity.kt (hunk 3 above)
Comment (ShobDeloitte) [reply app/src/main/java/com/dsu/pocapplication/news/ui/login/activity/LoginActivity.kt]: done

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/fragments/LoginFragment.kt (hunk 4 above)
Comment (ShobDeloitte) [reply app/src/main/java/com/dsu/pocapplication/news/ui/login/fragments/LoginFragment.kt]: done

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/viewmodel/LoginViewModel.kt (hunk 6 above)
Comment (ShobDeloitte) [reply app/src/main/java/com/dsu/pocapplication/news/ui/login/viewmodel/LoginViewModel.kt]: done

File: app/src/main/java/com/dsu/pocapplication/news/repository/LoginRepository.kt [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.pocapplication.news.repository
+
+class LoginRepository {
+}
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/pocapplication/news/repository/LoginRepository.kt]: Loginrepository should not be a part of news package. Please keep the packages news, login inside the package 'repository'

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/LoginFragment.kt [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.pocapplication.news.ui.login
+
+class LoginFragment {
+}
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/pocapplication/news/ui/login/LoginFragment.kt]: Both activity file and fragment file shoule not be required. One among the two should do. Please decide and delete the respective file. If you are confused on activities and fragments, you can explore on the same. Here is a start: https://stackoverflow.com/questions/10478233/why-fragments-and-when-to-use-fragments-instead-of-activities

**Review by asdheeraj — COMMENTED** [review]
For now, we can see the package structure for Login functionality. Please ensure the package structure for news is also uploaded in this same PR. (For repo and ui)

**Comment by asdheeraj** [comment]
Also, the commits are not yet squashed. Please do squash all your commits into a single commit as mentioned above. You can refer the link provided in the previous comment for reference

File: app/src/main/java/com/dsu/pocapplication/news/repository/LoginRepository.kt (hunk 7 above)
Comment (ShobDeloitte) [reply app/src/main/java/com/dsu/pocapplication/news/repository/LoginRepository.kt]: done

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/LoginFragment.kt (hunk 8 above)
Comment (ShobDeloitte) [reply app/src/main/java/com/dsu/pocapplication/news/ui/login/LoginFragment.kt]: done

File: app/src/main/java/com/dsu/pocapplication/news/ui/login/activity/LoginActivity.kt (hunk 3 above)
Comment (asdheeraj) [reply app/src/main/java/com/dsu/pocapplication/news/ui/login/activity/LoginActivity.kt]: Login should not be a part of the news package

Comment (asdheeraj) [reply app/src/main/java/com/dsu/pocapplication/news/ui/login/activity/LoginActivity.kt]: This was the similar comment ive given to repository. it should be addressed here as well

**Comment by asdheeraj** [comment]
The commits are not squashed still. Please resolve the conflicts as well as mentioned in the previous comment

File: app/src/main/java/com/dsu/poc/project/repository/login/LoginRepository.kt [hunk 9]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.poc.project.repository
+
+class LoginRepository {
+}
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/poc/project/repository/login/LoginRepository.kt]: project is not required. Please directly use the ui package. We know that the entire code is a part of a single project.

File: app/src/main/java/com/dsu/poc/project/ui/login/LoginActivity.kt [hunk 10]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.poc.project.ui.login
```
Comment (asdheeraj) [inline app/src/main/java/com/dsu/poc/project/ui/login/LoginActivity.kt]: This has to be a fragment as discussed. Please maintain a Parent activity with a reasonable name.

**Review by asdheeraj — COMMENTED** [review]
Please address the comments. I still see a package only for login. I still don't see a similar package for news. It has to be updated in the same PR. this was discussed on the call. Please address it.
Also, squash these commits into a single commit and resolve these conflicts as mentioned repeatedly in the previous comments.

**Review by asdheeraj — COMMENTED** [review]
Please reply me regarding these comments which are repeatedly put up in the previous comments:
1. Why are the commits not being squashed? I still can see three commits. Ur branch should only be one commit above development. I have mentioned this earlier.
image

2. The structure contains only the login functionality. There is no news structure similar to the news structure which I have mentioned in my previous two comments. 
image

3. Have you resolved all the conflicts? Also mentioned in my previous comments
image

The PR can't be merged until these comments are addressed. Please address them.

File: app/src/main/java/com/dsu/poc/project/ui/login/LoginActivity.kt (hunk 10 above)
Comment (ShobDeloitte) [reply app/src/main/java/com/dsu/poc/project/ui/login/LoginActivity.kt]: changed.

File: app/src/main/java/com/dsu/poc/project/repository/login/LoginRepository.kt (hunk 9 above)
Comment (ShobDeloitte) [reply app/src/main/java/com/dsu/poc/project/repository/login/LoginRepository.kt]: changed.

**Comment by asdheeraj** [comment]
When you are squashing ur commits, please rebase ur branch to development. **Always, ur current branch should be rebased to development.**(The branch into which ur PR is to be merged) That is how you will resolve conflicts

File: app/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+    <application
+            android:allowBackup="true"
+            android:icon="@mipmap/ic_launcher"
+            android:label="@string/app_name"
+            android:roundIcon="@mipmap/ic_launcher_round"
+            android:supportsRtl="true"
+            android:theme="@style/AppTheme">
+
+        <activity android:name=".base.BaseActivity">
```
Comment (debasish1987) [inline app/src/main/AndroidManifest.xml]: BaseActivity should not be a launcher activity, it should be an abstract class

File: app/src/main/java/com/dsu/poc/base/BaseActivity.kt [hunk 12]
```diff
@@ -0,0 +1,6 @@
+package com.dsu.poc.base
+
+import androidx.appcompat.app.AppCompatActivity
+
+class BaseActivity : AppCompatActivity() {
```
Comment (debasish1987) [inline app/src/main/java/com/dsu/poc/base/BaseActivity.kt]: it should be an abstract class

File: app/src/main/java/com/dsu/poc/base/BaseFragment.kt [hunk 13]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.poc.base
+
+class BaseFragment {
```
Comment (debasish1987) [inline app/src/main/java/com/dsu/poc/base/BaseFragment.kt]: it should be an abstract class

File: app/src/main/java/com/dsu/poc/repository/login/LoginRepository.kt [hunk 14]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.poc.project.repository
+
+class LoginRepository {
```
Comment (debasish1987) [inline app/src/main/java/com/dsu/poc/repository/login/LoginRepository.kt]: it should be interface

File: app/src/main/java/com/dsu/poc/repository/news/NewsRepository.kt [hunk 15]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.poc.repository.news
+
+class NewsRepository {
```
Comment (debasish1987) [inline app/src/main/java/com/dsu/poc/repository/news/NewsRepository.kt]: it should be interface

File: app/src/main/java/com/dsu/poc/ui/login/LoginFragment.kt [hunk 16]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.poc.project.ui.login
+
+class LoginFragment {
```
Comment (debasish1987) [inline app/src/main/java/com/dsu/poc/ui/login/LoginFragment.kt]: extend to baseFragment

File: app/src/main/java/com/dsu/poc/ui/news/NewsFragment.kt [hunk 17]
```diff
@@ -0,0 +1,4 @@
+package com.dsu.poc.ui.news
+
+class NewsFragment {
```
Comment (debasish1987) [inline app/src/main/java/com/dsu/poc/ui/news/NewsFragment.kt:3]: extend to baseFragment

**Review by debasish1987 — COMMENTED** [review]
java docs missing

**Review by debasish1987 — APPROVED** [review]

**Review by asdheeraj — COMMENTED** [review]
Please squash ur commits

**Review by asdheeraj — APPROVED** [review]

**Pull request merged**